### PR TITLE
2.0 SPCR-254 New add person form

### PR DESCRIPTION
**Pull request by jennikate**
## Description

- setup new people section in pages
- add new people multipage form
- move person form validation to people section
- make v2 component folder 
- make FormFieldError component for v2 page needs
- make v2 apiHooks for v2 page needs
- point links to new people form

NOT IN THIS PR
- editing people

To test

- go to /people
- click on save new person
- click on continue
- > all fields should error
- fill in date of birth with a date in the future
- click continue
- > date of birth field should error as invalid date
- fill in date of birth with an invalid date (e.g. month=22)
- > date of birth field should error as invalid date
- fill in other fields
- click continue
- > you should be taken to page 2 of the form
- go back to page 1
- > data should be persisted in the fields
- click continue
- > you should be taken to page 2 of the form
- click submit
- > all fields should error
- fill in an invalid expiry date (e.g. month=22)
- click submit
- > expiry date should error as invalid date
- enter an expiry date in the past
- > expiry date should error as an invalid date
- select document type of `Another travel document`
- > a free text field should be displayed
- click submit 
- > document type should show error
- enter text in the please specify field
- click submit
- > document type should NOT error
- change document type to passport
- enter document issuing state in lowercase letters (e.g. gbr)
- click submit
- > document issuing state should not error
- fill in the rest of the fields
- click submit
- > you should be returned to the people page with the list of people
- > your person should be visible in that list (in alphabetical order surname>firstname)

## Developer Checklist

\* Required

- [ ] Up-to-date with master branch? \*

- [ ] Does it have tests?

- [ ] Pull request URL linked to JIRA ticket? \*

- [ ] All reviewer comments resolved/answered? \*

Ensure you delete the branch once the Pull Request is merged.

### CODE REVIEW

File: src/pages/people/PersonForm.jsx [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+      lastName: formData.lastName,
+      documentType: formData.documentType,
+      documentNumber: formData.documentNumber,
+      documentIssuingState: formData.documentIssuingState.toUpperCase(),
+      documentExpiryDate: formatDate(formData.documentExpiryDateYear, formData.documentExpiryDateMonth, formData.documentExpiryDateDay),
+      dateOfBirth: formatDate(formData.dateOfBirthYear, formData.dateOfBirthMonth, formData.dateOfBirthDay),
+      nationality: formData.nationality,
+      placeOfBirth: formData.placeOfBirth,
+      gender: formData.gender,
+      peopleType: 'Crew',
```
Comment (Adele-B) [inline src/pages/people/PersonForm.jsx:103]: This seems to default the added person to employed crew

File: src/pages/people/PersonForm.jsx [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+                      onChange={handleChange}
+                    />
+                  </div>
+                  <div id="nationality" className={`govuk-form-group ${errors.nationality ? 'govuk-form-group--error' : ''}`}>
+                    <label className="govuk-label" htmlFor="nationality">
+                      Nationality
+                    </label>
+                    <FormFieldError error={errors.nationality} />
+                    <select
+                      data-testId="nationality-field"
```
Comment (Adele-B) [inline src/pages/people/PersonForm.jsx:518]: Small typo - the id in `data-testId` should be lowercase

**Review by Adele-B — CHANGES_REQUESTED** [review]

File: src/pages/people/PersonForm.jsx (hunk 1 above)
Comment (jennikate) [reply src/pages/people/PersonForm.jsx:103]: That's correct, that's what we want it to do. The user will be able to change the peopleType on the 2.x voyage form

Comment (Adele-B) [reply src/pages/people/PersonForm.jsx:103]: Ah my mistake, I thought the default was going to be unpaid crew, ignore me! 😅

**Review by Adele-B — APPROVED** [review]
LGTM

**Pull request merged**